### ISSUE TITLE
Make it easier to "fade" Hidden component

### ISSUE THREAD

**Issue by diegohaz** [diegohaz/reas#57]
Currently, if we want to "fade" or apply another transitions to `Hidden` and derivate components, we have to do this:
```js
const MyHidden = Hidden.extend`
  display: block !important;
  opacity: 0;
  ${props => props.visible && "opacity: 1"};
`;
```

We need a nicer way to do that, without having to put that `!important`.

**Comment by wordofchristian** [comment]
What would be the preference here? A prop on Hidden that specifies transition time? Or maybe a separate `<Fade />` component [similar to material-ui](https://material-ui.com/utils/transitions/).

**Comment by diegohaz** [comment]
I'm not sure about what approach to take here.

This is the current Hidden style:
https://github.com/diegohaz/reas/blob/019b901fa934a3c599d1a5b70772db760a40d46a/src/components/Hidden/Hidden.js#L43-L49

I was thinking about some prop that, instead of applying `display: none;`, just modified `opacity`, like the styles I posted.
```jsx
<Hidden fade />
```

**Comment by wordofchristian** [comment]
That looks good.

My only concern is what the escape hatch would be for extensibility. I think the ability to hide/show something is universally useful but animation timings can be very personal and different in every app.

I would hate for people to be in the position where they have to give up on using the built in fade because it doesn't quite match the look/feel they want.

**Comment by diegohaz** [comment]
I agree. I wasn't thinking about dealing with transitions, just `opacity`.

```jsx
<Hidden /> // changes display
<Hidden destroy /> // removes from the DOM
<Hidden fade /> // changes opacity instead of display
```

That would cover the three most common techniques to hide an element.

`fade` might be a bad name since it can infer animation, but I can't think any better. Maybe an antonym of `opaque`.

**Comment by diegohaz** [comment]
I was able to make some fancy stuff with `Step`, which uses `Hidden` underneath, playing around with `opacity`: https://codesandbox.io/s/4090w91mq0

**Comment by wordofchristian** [comment]
Ah! thank you for clarifying. That makes more sense now. 

I would say the best word to use would `invisible` or `transparent`.

We also might want to consider using `visibility: hidden` for the implementation rather than `opacity: 0`. The main difference being that an element with `opacity: 0` will still intercept click events whereas an element with `visibility: hidden` will not.  See [chart](https://stackoverflow.com/a/34529598/80559). We could also support all these options:

```javascript
<Hidden /> // changes to display: none
<Hidden destroy /> // removes from the DOM
<Hidden invisible /> // changes to visibility: hidden
<Hidden transparent /> // changes to opacity: 0
```

**Comment by diegohaz** [comment]
I've completely forgotten `visibility: hidden`. You're right. 

However, it doesn't look good hahah
```jsx
<Hidden invisible visible />
```

This conversation makes me think that if we need to read comments to understand what a prop does, we should find better names. So probably something more concrete:
```jsx
<Hidden />
<Hidden useDOM />
<Hidden useVisibility />
<Hidden useOpacity />
```

**Comment by wordofchristian** [comment]
I think this is the right direction, but I think `useDOM` is still somewhat unclear. Maybe it should be something like `useRemoval` or `useConditionalRendering`? I'm still not sure, but I think the name should imply something about what will actually happen to the DOM.
